ui/pages/confirmations/hooks/alerts/useBlockaidAlerts.ts

```diff
@@ -49,21 +49,27 @@ type SecurityAlertResponsesState = {
 
 const useBlockaidAlerts = (): Alert[] => {
   const t = useI18nContext();
-
-  const currentConfirmation = useSelector(
-    currentConfirmationSelector,
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  ) as Record<string, any>;
-
+  const { currentConfirmation } = useConfirmContext();
   const selectorChainId = useSelector(getCurrentChainId);
 
-  const securityAlertId = currentConfirmation?.securityAlertResponse
-    ?.securityAlertId as string;
+  const securityAlertId = (
+    currentConfirmation?.securityAlertResponse as SecurityAlertResponse
+  )?.securityAlertId as string;
 
   const transactionType = currentConfirmation?.type as TransactionType;
 
   const signatureSecurityAlertResponse = useSelector(
-    currentSignatureRequestSecurityResponseSelector,
```

Comment:
I suppose this is one disadvantage of using a context, as we can't use selectors to modularise how to extract different data from it.

Though I suppose if we had lots of re-use, we could instead create hooks such as `useCurrentSignatureSecurityResponse`?